### PR TITLE
Create editing and adding modals for coach editor

### PR DESCRIPTION
**Pull request by ExxZzyy**
Create editing and adding modals for coach editor

### CODE REVIEW

File: src/app/models/coach-profile/coach-profile-editor/coach-profile-editor-grammar-create-dialog/coach-profile-editor-grammar-create-dialog.component.html [hunk 1]
```diff
@@ -0,0 +1,25 @@
+<div class="container">
```
Comment (RomPavel) [inline src/app/models/coach-profile/coach-profile-editor/coach-profile-editor-grammar-create-dialog/coach-profile-editor-grammar-create-dialog.component.html:1]: file name is too long, lets leave only create-grammar-dialog.component.html, same for topic (same for .scss .ts)

File: src/app/models/coach-profile/coach-profile-editor/coach-profile-editor-grammar-create-dialog/coach-profile-editor-grammar-create-dialog.component.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+  levels: Level[] = [
+    Level.Advanced,
+    Level.UpperIntermediate,
+    Level.Intermediate,
+    Level.PreIntermediate,
+    Level.Elementary,
+  ];
+
+  questions: GrammarQuestions[] = [
```
Comment (RomPavel) [inline src/app/models/coach-profile/coach-profile-editor/coach-profile-editor-grammar-create-dialog/coach-profile-editor-grammar-create-dialog.component.ts:22]: I'am pretty sure its not a question, its answer

File: src/app/models/coach-profile/coach-profile-editor/coach-profile-editor-grammar-create-dialog/coach-profile-editor-grammar-create-dialog.component.ts [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+@Component({
+  selector: 'app-coach-profile-editor-grammar-dialog',
+  templateUrl: './coach-profile-editor-grammar-create-dialog.component.html',
+  styleUrls: ['./coach-profile-editor-grammar-create-dialog.component.scss'],
+})
+export class CoachProfileEditorGrammarCreateDialogComponent {
+  constructor(@Inject(MAT_DIALOG_DATA) public data: DialogData) {}
+
+  levels: Level[] = [
```
Comment (Inunor) [inline src/app/models/coach-profile/coach-profile-editor/coach-profile-editor-grammar-create-dialog/coach-profile-editor-grammar-create-dialog.component.ts:14]: I think you can convert it somehow. Something like Object.values(Level) or something. I don't like the idea about manually type all Level enum values: if you will add new Levels you will need to update this array

File: src/app/models/coach-profile/coach-profile-editor/coach-profile-editor-grammar-create-dialog/coach-profile-editor-grammar-create-dialog.component.ts [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+    Level.Advanced,
+    Level.UpperIntermediate,
+    Level.Intermediate,
+    Level.PreIntermediate,
+    Level.Elementary,
+  ];
+
+  questions: GrammarQuestions[] = [
+    GrammarQuestions.first,
```
Comment (Inunor) [inline src/app/models/coach-profile/coach-profile-editor/coach-profile-editor-grammar-create-dialog/coach-profile-editor-grammar-create-dialog.component.ts:23]: Same here as I understand

File: src/app/models/coach-profile/coach-profile-editor/coach-profile-editor-grammar-create-dialog/coach-profile-editor-grammar-create-dialog.component.html (hunk 1 above)
Comment (ExxZzyy) [reply src/app/models/coach-profile/coach-profile-editor/coach-profile-editor-grammar-create-dialog/coach-profile-editor-grammar-create-dialog.component.html]: Done 9f05d81454e778a2629b401adeafae98cdd42344

File: src/app/models/coach-profile/coach-profile-editor/coach-profile-editor-grammar-create-dialog/coach-profile-editor-grammar-create-dialog.component.ts (hunk 2 above)
Comment (ExxZzyy) [reply src/app/models/coach-profile/coach-profile-editor/coach-profile-editor-grammar-create-dialog/coach-profile-editor-grammar-create-dialog.component.ts]: Done 9f05d81454e778a2629b401adeafae98cdd42344

File: src/app/models/coach-profile/coach-profile-editor/grammar-adding-editing-dialog/grammar-adding-editing-dialog.component.ts [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+  constructor(@Inject(MAT_DIALOG_DATA) public data: CreateDialogData | EditDialogData) {
+    if (data.isEdit) {
+      this.grammar = {
+        answers: data.answers,
+        question: data.question,
+      };
+    }
+  }
+
+  levels: Level[] = [
```
Comment (RomPavel) [inline src/app/models/coach-profile/coach-profile-editor/grammar-adding-editing-dialog/grammar-adding-editing-dialog.component.ts:64]: remove this

File: src/app/models/coach-profile/coach-profile-editor/topic-adding-editing-dialog/topic-adding-editing-dialog.component.ts [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+
+  constructor(@Inject(MAT_DIALOG_DATA) public data: CreateDialogData | EditDialogData) {
+    if (data.isEdit) {
+      this.topic = {
+        question: data.question,
+      };
+    }
+  }
+
+  levels: Level[] = [
```
Comment (RomPavel) [inline src/app/models/coach-profile/coach-profile-editor/topic-adding-editing-dialog/topic-adding-editing-dialog.component.ts:33]: remove this

File: src/mocks/users-utils.mock.ts [hunk 7]
```diff
@@ -12,11 +12,30 @@ export interface TestData {
...
   id: string;
   level: string;
 }
 
+export interface DialogData {
+  id: number;
+  level: string;
+}
+
+export interface Level {
```
Comment (RomPavel) [inline src/mocks/users-utils.mock.ts:35]: Do you still need it

File: src/mocks/users-utils.mock.ts [hunk 8]
```diff
@@ -12,11 +12,30 @@ export interface TestData {
   date: Date;
 }
 
+// export interface CoachEditorTest {
```
Comment (RomPavel) [inline src/mocks/users-utils.mock.ts:15]: remove this

Comment (ExxZzyy) [reply src/mocks/users-utils.mock.ts]: Done 1bd925c7a81b64a5065b3469e412ebca208d6a5f

File: src/mocks/users-utils.mock.ts (hunk 7 above)
Comment (ExxZzyy) [reply src/mocks/users-utils.mock.ts]: No, deleted 1bd925c7a81b64a5065b3469e412ebca208d6a5f

File: src/app/models/coach-profile/coach-profile-editor/topic-adding-editing-dialog/topic-adding-editing-dialog.component.ts (hunk 6 above)
Comment (ExxZzyy) [reply src/app/models/coach-profile/coach-profile-editor/topic-adding-editing-dialog/topic-adding-editing-dialog.component.ts]: Done 1bd925c7a81b64a5065b3469e412ebca208d6a5f

File: src/app/models/coach-profile/coach-profile-editor/grammar-adding-editing-dialog/grammar-adding-editing-dialog.component.ts (hunk 5 above)
Comment (ExxZzyy) [reply src/app/models/coach-profile/coach-profile-editor/grammar-adding-editing-dialog/grammar-adding-editing-dialog.component.ts]: Done 1bd925c7a81b64a5065b3469e412ebca208d6a5f

File: src/app/models/coach-profile/coach-profile-editor/coach-profile-editor-grammar-create-dialog/coach-profile-editor-grammar-create-dialog.component.ts (hunk 3 above)
Comment (ExxZzyy) [reply src/app/models/coach-profile/coach-profile-editor/coach-profile-editor-grammar-create-dialog/coach-profile-editor-grammar-create-dialog.component.ts]: Done 1bd925c7a81b64a5065b3469e412ebca208d6a5f

File: src/app/models/coach-profile/coach-profile-editor/coach-profile-editor-grammar-create-dialog/coach-profile-editor-grammar-create-dialog.component.ts (hunk 4 above)
Comment (ExxZzyy) [reply src/app/models/coach-profile/coach-profile-editor/coach-profile-editor-grammar-create-dialog/coach-profile-editor-grammar-create-dialog.component.ts]: We have fixed number of answers, is it really needed?

File: src/app/components/choose-level/choose-level.component.ts [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+  selector: 'app-choose-level',
+  templateUrl: './choose-level.component.html',
+  styleUrls: ['./choose-level.component.scss'],
+})
+export class ChooseLevelComponent {
+  @Input() currentLevel: Level | undefined;
+
+  @Output() changeLevel = new EventEmitter<Level>();
+
+  levels: Level[] = [
```
Comment (RomPavel) [inline src/app/components/choose-level/choose-level.component.ts:15]: As @Inunor said, lets create constant somewhere near **Level** enum
like **const LevelList = Object.values(Level);** and lets reuse it here

**Review by RomPavel — APPROVED** [review]

File: src/app/models/coach-profile/coach-profile-editor/coach-profile-editor-grammar-create-dialog/coach-profile-editor-grammar-create-dialog.component.ts (hunk 4 above)
Comment (ExxZzyy) [reply src/app/models/coach-profile/coach-profile-editor/coach-profile-editor-grammar-create-dialog/coach-profile-editor-grammar-create-dialog.component.ts]: Done 5ef4d745798893da8ad068cafcc184bc0d879306

File: src/app/components/choose-level/choose-level.component.ts (hunk 9 above)
Comment (ExxZzyy) [reply src/app/components/choose-level/choose-level.component.ts]: Done 5ef4d745798893da8ad068cafcc184bc0d879306

**Pull request merged**